sqlx-core/src/postgres/types/ipaddr.rs

```diff
@@ -0,0 +1,54 @@
+use std::net::IpAddr;
+
+use ipnetwork::IpNetwork;
```

Comment:
This probably shouldn't use the `ipnetwork` crate; in fact, I'd almost say that this should be where the actual implementation lives and then the `ipnetwork` module can be a shim around this.

Reply:
`inet` and `cidr` have a subnet mask too, `IpNetwork` got that, `IpAddr` don't.

Reply:
I was wandering this too when I saw this PR and altough I haven't looked if this is already being done with other types, the idea of lossy data decoding doesn't sound like a great thing to have, unless we only allow it on types where a constraint limits the mask to /32 for IPv4, /128 for IPv6? :thinking:

Reply:
In those cases the subnet would be `0` as far as I know.

Reply:
Nope, I'm dumb, added the decoding with error if the prefix is not a single IP address, which is what inet/cidr default to when no prefix is defined.